packages/core/src/utils/result.ts

```diff
@@ -146,8 +150,11 @@ export const mergeResultPatch = (
       }
     }
   } else {
-    withData.data = nextResult.data || prevResult.data;
-    errors = (nextResult.errors as any[]) || errors;
+    withData.data = (nextResult.payload || nextResult).data || prevResult.data;
+    errors =
```

Comment:
In addition to GraphQL errors there can also be fatal transport-level errors:

> If a fatal transport-level error occurs, the router sends a message with a top-level errors field and null payload field, then closes the connection

We opted to [stash protocol errors in `extensions`](https://github.com/apollographql/apollo-client/pull/10592/files#diff-d180d739ac5085e3556db4fb80641f30389c9958f0eb702e88475d6253506e18R88-R96) (using a symbol property to avoid possible collisions) to pass the error through to client internals and then handle them where we process subscription errors.

Reply:
Hmm, is there a need for that? As essentially it will deliver the errors on that property and there is no possibility for an accompanying payload. I guess a recipe we should look into is re-establishing the connection if that occurs due to i.e. an interrupted connection

Reply:
If this is desirable, we can throw an error in here as well, which will, in the `fetchExchange` at least, be caught here: https://github.com/urql-graphql/urql/blob/5b6449e62d8c3eb7f0f863ae98b3994871c2da38/packages/core/src/internal/fetchSource.ts#L213

i.e. be wrapped in a `CombinedError`.

If this is a protocol-level error, that's the right place to put it, I'd say, if it necessitates destroying the response entirely.

Just to clarify this _will_ make prior data, errors, and extensions irrecoverable in most cases for the UI bindings and following exchanges handling results.

However, the reason why this is potentially problematic is:
- `errors` mirrors the GraphQL result spec and can hence be an array and isn't limited, i.e. `path` and such may still be provided
- throwing is only acceptable if `'payload' in result` hence it's hard to make a decision here without knowing more about protocol implementers

@alessbell: Thanks for pointing this out! 🙌 Is there any case where `payload: null` isn't set in a multipart subscription response and/or the root (not `.payload`) errors array will ever contain more than one protocol-level error that prohibits keeping the rest of the data around?

Reply:
@kitten: thanks for your patience here - we're in the process of clarifying one edge case that came up whereby a subgraph emits a heartbeat-like chunk of `payload: null` (without `errors`), which we may amend the spec text to account for. I should have a clearer answer in the next day or so, thanks!